scheduler/replica_scheduler.go

```diff
@@ -70,23 +71,61 @@ func (rcs *ReplicaScheduler) getRandomNode(nodeMap map[string]*longhorn.Node) *l
 	return node
 }
 
-func (rcs *ReplicaScheduler) preferredNodes(nodeInfo map[string]*longhorn.Node, replicas map[string]*longhorn.Replica) (map[string]*longhorn.Node, error) {
+func (rcs *ReplicaScheduler) preferredNodes(nodeInfo map[string]*longhorn.Node, replicas map[string]*longhorn.Replica, replica *longhorn.Replica) (map[string]*longhorn.Node, error) {
 	preferredNode := map[string]*longhorn.Node{}
+	filterdNode := []*longhorn.Node{}
 	for nodeName, node := range nodeInfo {
 		isFilterd := false
 		for _, r := range replicas {
-			if r.Spec.NodeID != "" && r.Spec.NodeID == nodeName {
+			// filter current replica when rebuild
```

Comment:
Um, I don't understand when this can be true. The rebuild of replica will start another replica, not reusing the same replica. When will the replica itself in the list?